File: DP(Dynamic_Programming, 동적계획법)/백준_9095_1,2,3더하기/1,2,3 더하기.py
arr = [0 for _ in range(11)]
arr[1] = 1; arr[2] = 2;  arr[3] = 4
T = int(input())
for i in range(T):
    n = int(input())
    if arr[n] > 0: print(arr[n])
    else:
        for j in range(4, n+1):
            arr[j] = arr[j-1] +arr[j-2] +arr[j-3]
        print(arr[n])

File: DP(Dynamic_Programming, 동적계획법)/백준_9095_1,2,3더하기/1,2,3더하기.cs
using System.Text;
using System;
using static System.Console;

class Program
{
    static int[] arr = new int[12];
    public static void Main(string[] args)
    {
        arr[1] = 1;
        arr[2] = 2;
        arr[3] = 4;
        int T = int.Parse(ReadLine());
        for(int i = 0; i< T; i++)
        {
            int n = int.Parse(ReadLine());
            if (arr[n] > 0) WriteLine(arr[n]);
            else
            {
                for(int j = 4; j<= n; j++)
                {
                    arr[j] = arr[j - 1] + arr[j - 2] + arr[j - 3];
                }
                WriteLine(arr[n]);
            }
        }


    }
}